manifests/director.pp

```diff
@@ -32,28 +33,20 @@
     }
   }
 
-  # directories
-  $config_director_dirs = [
-    $config_dir,
-    "${config_dir}/catalog",
-    "${config_dir}/client",
-    "${config_dir}/console",
-    "${config_dir}/counter",
-    "${config_dir}/director",
-    "${config_dir}/fileset",
-    "${config_dir}/job",
-    "${config_dir}/jobdefs",
-    "${config_dir}/messages",
-    "${config_dir}/pool",
-    "${config_dir}/profile",
-    "${config_dir}/schedule",
-    "${config_dir}/storage",
-  ]
+  file { $config_dir:
+    ensure  => directory,
+    force   => true,
+    mode    => $::bareos::file_dir_mode,
+    owner   => $::bareos::file_owner,
+    group   => $::bareos::file_group,
+    require => Package[$package_name],
+    notify  => Service[$service_name],
+    tag     => ['bareos', 'bareos_director'],
+  }
 
-  file { $config_director_dirs:
+  file { "${config_dir}/${managed_dirs}":
```

Comment:
This doesn't work. You can't concatenate a string with an array. You can use a loop:
```puppet
$managed_dirs.each |$managed_dir| {
  file { "${config_dir}/${managed_dir}":
    # ...
  }
}
```

Reply:
ahh that explains the weird error in CI, thanks fix that and tested it, works